Gordon360/Services/ProfileService.cs

```diff
@@ -323,6 +326,49 @@ public async Task<StudentProfileViewModel> UpdateMobilePhoneNumberAsync(string u
             return profile;
         }
 
+        /// <summary>
+        /// office location setting
+        /// </summary>
+        /// <param name="username"> The username for the user whose office location is to be updated </param>
+        /// <param name="newBuilding">The new building location to update the user's office location to</param> 
+        /// <param name="newRoom">The new room to update the user's office room to</param>
+        /// <returns>updated fac/staff profile if found</returns>
+        public async Task<FacultyStaffProfileViewModel> UpdateOfficeLocationAsync(string username, string newBuilding, string newRoom)
+        {
+            var profile = GetFacultyStaffProfileByUsername(username);
+            if (profile == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
+            }
+            var user = _webSQLContext.accounts.FirstOrDefault(a => a.AD_Username == username);
+            user.Building = newBuilding;
+            user.Room = newRoom;
+            await _webSQLContext.SaveChangesAsync();
+
+            return profile;
+        }
+
+        /// <summary>
+        /// office hours setting
+        /// </summary>
+        /// <param name="username"> The username for the user whose office hours is to be updated </param>
+        /// <param name="newHours">The new hours to update the user's office hours to</param>
+        /// <returns>updated fac/staff profile if found</returns>
+        public async Task<FacultyStaffProfileViewModel> UpdateOfficeHoursAsync(string username, string newHours)
+        {
+            var profile = GetFacultyStaffProfileByUsername(username);
+            if (profile == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
+            }
+            var acccount = _webSQLContext.accounts.FirstOrDefault(a => a.AD_Username == username);
+            var user = _webSQLContext.account_profiles.FirstOrDefault(a => a.account_id == acccount.account_id);
```

Comment:
```suggestion
            var acccount = _webSQLContext.accounts.FirstOrDefault(a => a.AD_Username == username);
            var user = _webSQLContext.account_profiles.FirstOrDefault(a => a.account_id == acccount.account_id);
```

this can be a join